wrappers/openvino/rs-vino/age-gender-detection.cpp

```diff
@@ -113,21 +128,78 @@ namespace openvino_helpers
 
         if (ptrAgeOutput->getCreatorLayer().lock()->type != "Convolution")
             throw std::logic_error("In Age/Gender Recognition network, age layer (" + ageCreatorLayer->name +
-                                    ") should be a Convolution, but was: " + ageCreatorLayer->type);
+                ") should be a Convolution, but was: " + ageCreatorLayer->type);
 
         if (ptrGenderOutput->getCreatorLayer().lock()->type != "SoftMax")
             throw std::logic_error("In Age/Gender Recognition network, gender layer (" + genderCreatorLayer->name +
-                                    ") should be a SoftMax, but was: " + genderCreatorLayer->type);
-        if( doRawOutputMessages )
+                ") should be a SoftMax, but was: " + genderCreatorLayer->type);
+
+        if (doRawOutputMessages)
         {
             LOG(DEBUG) << "Age layer: " << ageCreatorLayer->name;
             LOG(DEBUG) << "Gender layer: " << genderCreatorLayer->name;
         }
+#else
+#ifdef OPENVINO_NGRAPH
+        if (auto ngraphFunction = network.getFunction()) {
+
+                for (const auto& op : ngraphFunction->get_ops()) {
+                    std::string friendly_name = op->get_friendly_name();
+                    std::string output_type = op->get_type_name();
+
+                    if ((friendly_name.find(ptrGenderOutput->getName()) != std::string::npos) && (output_type == "Convolution")) {
+                         std::swap(ptrAgeOutput, ptrGenderOutput);
```

Comment:
Can you explain, please?

Reply:
looking for the age and gender nodes in the ngraph. The age layer node should be Convolution type. if found ptrGenderOutput is with Convolution type, swap them to ensure ptrAgeOutput is the one with Convolution type.